lmdeploy/pytorch/engine/engine.py

```diff
@@ -636,7 +631,11 @@ async def __long_context_single_forward(inputs):
                 ret['logits'] = ret['logits'][:, last_token_loc]
             return ret
         else:
-            return await __long_context_single_forward(inputs)
+            ret = await __long_context_single_forward(inputs)
+            if not return_logits and not inputs.is_decoding:
```

Comment:
what benefits can we get from this change?

Reply:
https://github.com/InternLM/lmdeploy/blob/9cb25830bf61a86fbf4bf4a4de801ecea4a38d31/lmdeploy/pytorch/engine/engine.py#L518-L525